eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/RelationConverter.php

```diff
@@ -66,30 +67,101 @@ public function toFieldValue(StorageFieldValue $value, FieldValue $fieldValue)
      */
     public function toStorageFieldDefinition(FieldDefinition $fieldDef, StorageFieldDefinition $storageDef)
     {
-        // Selection method, 0 = browse, 1 = dropdown
-        $storageDef->dataInt1 = (int)$fieldDef->fieldTypeConstraints->fieldSettings['selectionMethod'];
+        $fieldSettings = $fieldDef->fieldTypeConstraints->fieldSettings;
+        $doc = new DOMDocument('1.0', 'utf-8');
+        $root = $doc->createElement('related-objects');
+        $doc->appendChild($root);
 
-        // Selection root, location ID
-        $storageDef->dataInt2 = (int)$fieldDef->fieldTypeConstraints->fieldSettings['selectionRoot'];
+        $constraints = $doc->createElement('constraints');
+        if (!empty($fieldSettings['selectionContentTypes'])) {
+            foreach ($fieldSettings['selectionContentTypes'] as $typeIdentifier) {
+                $allowedClass = $doc->createElement('allowed-class');
+                $allowedClass->setAttribute('contentclass-identifier', $typeIdentifier);
+                $constraints->appendChild($allowedClass);
+                unset($allowedClass);
+            }
+        }
+        $root->appendChild($constraints);
+
+        $selectionType = $doc->createElement('selection_type');
+        if (isset($fieldSettings['selectionMethod'])) {
+            $selectionType->setAttribute('value', (int)$fieldSettings['selectionMethod']);
+        } else {
+            $selectionType->setAttribute('value', 0);
+        }
+        $root->appendChild($selectionType);
+
+        $defaultLocation = $doc->createElement('contentobject-placement');
+        if (!empty($fieldSettings['selectionRoot'])) {
+            $defaultLocation->setAttribute('node-id', (int)$fieldSettings['selectionRoot']);
+        }
+        $root->appendChild($defaultLocation);
+
+        $doc->appendChild($root);
+        $storageDef->dataText5 = $doc->saveXML();
     }
 
     /**
      * Converts field definition data in $storageDef into $fieldDef.
      *
-     * @param \eZ\Publish\Core\Persistence\Legacy\Content\StorageFieldDefinition $storageDef
+     * <?xml version="1.0" encoding="utf-8"?>
+     * <related-objects>
+     *   <constraints>
+     *     <allowed-class contentclass-identifier="blog_post"/>
+     *   </constraints>
+     *   <selection_type value="1"/>
+     *   <contentobject-placement node-id="67"/>
+     * </related-objects>
+     *
+     * <?xml version="1.0" encoding="utf-8"?>
+     * <related-objects>
+     *   <constraints/>
+     *   <selection_type value="0"/>
+     *   <contentobject-placement/>
+     * </related-objects>
```

Comment:
> is to create Relation field limited to e.g. blog_post and try to create relation using different content type to check if it throws exception

It won't throw anything. On PHP level it creates anything happily. But this is the RelationList's behavior too. The setting is only affecting the JS validation atm.